### PR TITLE
Syndicate sponge for uplink

### PR DESCRIPTION
**Pull request by Arendian**
## About the PR

This PR adds an iteration of the syndicate sponge from TG to the uplink.
For 6 tc people can buy a box containing 6 syndicate sponges disguised as monkey cubes, upon getting hydrated these will randomly spawn one of the following mobs:
1. Rat servant (Controllable by ghosts)
2. Angry bee
3. Space Tick
4. Purple Snake
5. Space Spider (30% ghost role chance)
6. Space Kangaroo (20% ghost role chance)
7. Angry yellow slime (Controllable by ghosts)
8. Holocarp
9. SpaceBear
10. Xeno Ravager (Controllable by ghosts)

You are unable to pet the cubes so the wildlife will attack anyone in sight, including the person that spawned them.
Most mobs that open ghost roles already have rules telling them to be hostile to the crew, so that should prevent them from becoming friendly(I hope).

**Media**

https://github.com/space-wizards/space-station-14/assets/137322659/68c32887-f38b-4e96-8447-3d92c81a7f68

image
image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Dygon
- add: The Syndicate reverse engineered Drymate's monkey cube technology and has started capturing wildlife to turn into dried cubes. Syndicate agents can now buy Syndicate sponges from the uplink, these turn into random hostile wildlife upon getting wet.

### CODE REVIEW

File: Content.Server/Chemistry/EntitySystems/RehydratableSystem.cs [hunk 1]
```diff
@@ -30,7 +32,14 @@ private void Expand(EntityUid uid, RehydratableComponent comp)
...
 
-        var target = Spawn(comp.TargetPrototype, Transform(uid).Coordinates);
+        var target = new EntityUid();
+        var randomMob = _random.Next(0, 8);
+        string[] randomList = { "MobRatServant", "MobCarpHolo", "MobXenoRavager", "MobAngryBee", "MobAdultSlimesYellowAngry", "MobGiantSpiderAngry", "MobBearSpace", "MobPurpleSnake" };
+
+        if (comp.TargetPrototype == "Random")
+            target = Spawn(randomList[randomMob], Transform(uid).Coordinates);
+        else target = Spawn(comp.TargetPrototype, Transform(uid).Coordinates);
+
```
Comment (EmoGarbage404) [inline Content.Server/Chemistry/EntitySystems/RehydratableSystem.cs:42]: just change TargetPrototype to either be a list of strings or an entity spawn entry. You'll have to update the other instances of it but it'll be pretty easy.

Comment (Arendian) [reply Content.Server/Chemistry/EntitySystems/RehydratableSystem.cs]: Thanks for the tip, i'll try making the change you recommended.

File: Resources/Prototypes/Entities/Objects/Misc/monkeycube.yml [hunk 2]
```diff
@@ -35,3 +35,41 @@
   - type: Tag
     tags:
     - MonkeyCube
+
+- type: entity
+  parent: BoxCardboard
+  name: monkey cube box
```
Comment (PuroSlavKing) [inline Resources/Prototypes/Entities/Objects/Misc/monkeycube.yml:41]: maybe add "suffix: Syndie"

**Comment by Myakot** [comment]
Add #15744 into the pool for this to be more unique. (just a suggestion)

File: Content.Server/Chemistry/EntitySystems/RehydratableSystem.cs [hunk 3]
```diff
@@ -30,7 +32,10 @@ private void Expand(EntityUid uid, RehydratableComponent comp)
     {
         _popups.PopupEntity(Loc.GetString("rehydratable-component-expands-message", ("owner", uid)), uid);
 
-        var target = Spawn(comp.TargetPrototype, Transform(uid).Coordinates);
+        var randomMob = _random.Next(0, comp.TargetPrototype.Count);
+
+        var target = Spawn(comp.TargetPrototype[randomMob], Transform(uid).Coordinates);
```
Comment (EmoGarbage404) [inline Content.Server/Chemistry/EntitySystems/RehydratableSystem.cs:37]:
```suggestion
        var randomMob = _random.Pick(comp.TargetPrototype);

        var target = Spawn(randomMob, Transform(uid).Coordinates);
```

Comment (deltanedas) [reply Content.Server/Chemistry/EntitySystems/RehydratableSystem.cs:37]: also i dont think TargetPrototype is a good name when its multiple possible targets

Comment (Arendian) [reply Content.Server/Chemistry/EntitySystems/RehydratableSystem.cs]: I replaced the randomize code with the one EmoGarbage404 proposed and renamed TargetPrototype to PossibleSpawns

File: Resources/Prototypes/Entities/Objects/Misc/monkeycube.yml (hunk 2 above)
Comment (Arendian) [reply Resources/Prototypes/Entities/Objects/Misc/monkeycube.yml:41]: I added the "Syndicate" suffix to the monkey cube box, wrapped monkey cubes and monkey cubes.

**Comment by Arendian** [comment]
> Add #15744 into the pool for this to be more unique. (just a suggestion)

I added space ticks, space spiders and space kangaroos to the spawn pool

**Pull request merged**